### PR TITLE
Powermax operator support wizard

### PR DESCRIPTION
**Pull request by boyamurthy**
Adding operator support for Powermax in installation wizard 

# GitHub Issues
List the GitHub issues impacted by this PR:

| GitHub Issue # |
| -------------- |
| https://github.com/dell/csm/issues/885|

# Checklist:

- [X] Have you run a grammar and spell checks against your submission?
- [ ] Have you tested the changes locally?
- [ ] Have you tested whether the hyperlinks are working properly?
- [ ] Did you add the examples wherever applicable?
- [ ] Have you added high-resolution images?

### CODE REVIEW

**Review by sharont58 — DISMISSED** [review]

File: content/docs/deployment/csminstallationwizard/src/static/js/ui-functions.js [hunk 1]
```diff
@@ -276,10 +320,35 @@ function displayModules(installationType, driverName, CONSTANTS_PARAM) {
...
+			$(".resiliency").hide();
 			$(".migration").show();
 			$(".vSphere").show();
 			$(".storage-capacity").show();
+			$(".snapshot-feature").show();
+			$(".replication-mod").show();
+			$(".iscsichap").show();
+			$(".transport-protocol").show();
 			document.getElementById("driver-namespace").value = CONSTANTS_PARAM.POWERMAX_NAMESPACE;
+			if (installationType === 'operator') {
```
Comment (niranjan-n1) [inline content/docs/deployment/csminstallationwizard/src/static/js/ui-functions.js:332]:
```suggestion
			if (installationType === CONSTANTS_PARAM.OPERATOR) {
```

File: content/docs/deployment/csminstallationwizard/src/index.html [hunk 2]
```diff
@@ -368,6 +427,38 @@
...
+                      <div class="col-sm-3">
+                        <input class="form-control input-lg" type="text" id="transport-protocol" placeholder="Example: ISCSI ,FC or FIBRE " required>
+                      </div>
+                    </div>
+                  </div>
+
+		  <div class="row mb-4 iscsichap">
+                    <div class="col-sm-12">
+                      <input class="form-check-input" type="checkbox" id="iscsichap" value="" >
+                      <label for="iscsichap" class="form-check-label ms-2 text-dark" data-bs-toggle="tooltip" data-bs-placement="right" title="Enable/Disable ISCSI CHAPfeature">Iscsichap</label>
```
Comment (niranjan-n1) [inline content/docs/deployment/csminstallationwizard/src/index.html:443]:
```suggestion
                      <label for="iscsichap" class="form-check-label ms-2 text-dark" data-bs-toggle="tooltip" data-bs-placement="right" title="Enable/Disable ISCSI CHAPfeature">iSCSI CHAP</label>
```

File: content/docs/deployment/csminstallationwizard/src/static/js/generate-yaml.js [hunk 3]
```diff
@@ -147,8 +164,11 @@ function createYamlString(yamlTpl, yamlTplValues, driverParam, CONSTANTS_PARAM)
 	yamlTpl = yamlTpl.replaceAll("$POWERMAX_STORAGE_ARRAY_ENDPOINT_URL", yamlTplValues.storageArrayEndpointUrl);
 	yamlTpl = yamlTpl.replaceAll("$POWERMAX_STORAGE_ARRAY_BACKUP_ENDPOINT_URL", yamlTplValues.storageArrayBackupEndpointUrl);
 	yamlTpl = yamlTpl.replaceAll("$POWERMAX_MANAGEMENT_SERVERS_ENDPOINT_URL", yamlTplValues.storageArrayEndpointUrl);
+	yamlTpl = yamlTpl.replaceAll("$POWERMAX_MANAGE_ARRAY_ID", yamlTplValues.manageArrayId);
+	yamlTpl = yamlTpl.replaceAll("$POWERMAX_MANAGEE_ARRAY_ENDPOINT_URL", yamlTplValues.manageArrayEndpointUrl);
```
Comment (niranjan-n1) [inline content/docs/deployment/csminstallationwizard/src/static/js/generate-yaml.js:168]: There is a typo in POWERMAX_MANAGEE_ARRAY_ENDPOINT_URL

File: content/docs/deployment/csminstallationwizard/src/static/js/tests/generate-yaml.test.js [hunk 4]
```diff
@@ -101,6 +101,16 @@ describe("GIVEN setValues function", () => {
             <input type="checkbox" id="leader-election">
             <input type="text" id="authorization-proxy-host" value="">
             <input type="text" id="taint" value="node-role.kubernetes.io/control-plane">
+            <input type="text" id="target-cluster-id" value="">
+            <input type="text" id="transport-protocol" value="">
+            <input type="checkbox" id="iscsichap">
+            <input type="checkbox" id="observability-operator">
+            <input type="checkbox" id="observability-operator-metrics">
+            <input type="checkbox" id="vobservability-operator-topology">
```
Comment (niranjan-n1) [inline content/docs/deployment/csminstallationwizard/src/static/js/tests/generate-yaml.test.js:109]: There is a typo in the id "vobservability-operator-topology

File: content/docs/deployment/csminstallationwizard/src/static/js/tests/generate-yaml.test.js [hunk 5]
```diff
@@ -191,7 +211,18 @@ describe("GIVEN setValues function", () => {
             <input type="text" id="driver-pod-label" value="dell-storage">
             <input type="checkbox" id="volumeless-pods">
             <input type="checkbox" id="connection-validation">
+            <input type="text" id="target-cluster-id" value="">
+            <input type="text" id="transport-protocol" value="">
+            <input type="checkbox" id="iscsichap">
+            <input type="checkbox" id="observability-operator">
+            <input type="checkbox" id="observability-operator-metrics">
+            <input type="checkbox" id="vobservability-operator-topology">
```
Comment (niranjan-n1) [inline content/docs/deployment/csminstallationwizard/src/static/js/tests/generate-yaml.test.js:219]: There is a typo in the id "vobservability-operator-topology

File: content/docs/deployment/csminstallationwizard/src/index.html [hunk 6]
```diff
@@ -109,11 +109,50 @@
                     </div>
                   </div>
 
+		  <div class="row mb-4 observability-operator">
+                    <div class="col-sm-12">
+                      <input class="form-check-input" type="checkbox" id="observability-operator" value="" onchange="onObservabilityOperatorChange()">
```
Comment (niranjan-n1) [inline content/docs/deployment/csminstallationwizard/src/index.html:114]: Line number - 113 -132 can be removed and can reuse Line 100- 109 and onObservabilityChange() can updated to handle  onObservabilityOperatorChange()

**Review by niranjan-n1 — CHANGES_REQUESTED** [review]
Few comments and also Github actions are failing

File: content/docs/deployment/csminstallationwizard/src/static/js/ui-functions.js (hunk 1 above)
Comment (boyamurthy) [reply content/docs/deployment/csminstallationwizard/src/static/js/ui-functions.js]: Updated .

File: content/docs/deployment/csminstallationwizard/src/index.html (hunk 2 above)
Comment (boyamurthy) [reply content/docs/deployment/csminstallationwizard/src/index.html:443]: Updated .

File: content/docs/deployment/csminstallationwizard/src/static/js/generate-yaml.js (hunk 3 above)
Comment (boyamurthy) [reply content/docs/deployment/csminstallationwizard/src/static/js/generate-yaml.js:168]: fixed .

File: content/docs/deployment/csminstallationwizard/src/static/js/tests/generate-yaml.test.js (hunk 4 above)
Comment (boyamurthy) [reply content/docs/deployment/csminstallationwizard/src/static/js/tests/generate-yaml.test.js]: fixed.

File: content/docs/deployment/csminstallationwizard/src/static/js/tests/generate-yaml.test.js (hunk 5 above)
Comment (boyamurthy) [reply content/docs/deployment/csminstallationwizard/src/static/js/tests/generate-yaml.test.js]: fixed .

File: content/docs/deployment/csminstallationwizard/src/templates/operator/csm-powermax-1.7.0.template [hunk 7]
```diff
@@ -0,0 +1,318 @@
...
+        # Default value: None
+        # Examples: "000000000001", "000000000002"
+        - name: X_CSI_MANAGED_ARRAYS
+          value: $POWERMAX_MANAGE_ARRAY_ID
+        # X_CSI_POWERMAX_ENDPOINT: Address of the Unisphere server that is managing the PowerMax arrays
+        # In case of multi-array, provide an endpoint of locally attached array
+        # Default value: None
+        # Example: https://0.0.0.1:8443
+        - name: X_CSI_POWERMAX_ENDPOINT
+          value: $POWERMAX_MANAGEE_ARRAY_ENDPOINT_URL
```
[the remark this answers is not in the document]
Comment (boyamurthy) [reply content/docs/deployment/csminstallationwizard/src/templates/operator/csm-powermax-1.7.0.template]: fixed .

File: content/docs/deployment/csminstallationwizard/src/index.html (hunk 6 above)
Comment (boyamurthy) [reply content/docs/deployment/csminstallationwizard/src/index.html:114]: Will make these changes in separate PR .

**Review by niranjan-n1 — APPROVED** [review]
LGTM

**Review by shanmydell — APPROVED** [review]

**Review by samihan-dell — APPROVED** [review]

**Pull request merged**